scipy/fftpack/czt.py

```diff
@@ -0,0 +1,457 @@
+    w : complex, optional
+        The ratio between points in each step.
+    a : complex, optional
+        The starting point in the complex plane.  The default is 1.
+
+    Returns
+    -------
+    out : ndarray
+        The points in the Z plane at which the CZT samples the Z-transform,
+        as complex numbers.
+
+    Examples
+    --------
+    Plot the points of a 16-point FFT:
+
+    >>> import matplotlib.pyplot as plt
+    >>> points = czt_points(16)
+    >>> plt.plot(np.real(points), np.imag(points), 'o')
+    >>> plt.margins(0.1, 0.1); plt.axis('equal')
+    >>> plt.show()
+
+    and a 91-point logarithmic spiral:
+
+    >>> m, w, a = 91, 0.995*np.exp(-1j*np.pi*.05), 0.8*np.exp(1j*np.pi/6)
+    >>> points = czt_points(m, w, a)
+    >>> plt.plot(np.real(points), np.imag(points), 'o')
+    >>> plt.margins(0.1, 0.1); plt.axis('equal')
+    >>> plt.show()
+
+    """
+    m = _validate_sizes(1, m)
+
+    k = arange(m)
+
+    a = 1.0 * a  # at least float
+
+    if w is None:
+        # Nothing specified, default to FFT
+        return a * np.exp(2j * pi * k / m)
+    else:
+        # w specified
+        w = 1.0 * w  # at least float
+        return a * w**-k
+
+
+class CZT(object):
+    """
+    Create a chirp Z-transform function.
+
+    Transform to compute the frequency response around a spiral.
+    Objects of this class are callables which can compute the
+    chirp Z-transform on their inputs.  This object precalculates the constant
+    chirps used in the given transform.
+
+    Parameters
+    ----------
+    n : int
+        The size of the signal.
+    m : int, optional
+        The number of output points desired.  The default is `n`.
+    w : complex, optional
+        The ratio between points in each step.  This must be precise or the
+        accumulated error will degrade the tail of the output sequence.
+    a : complex, optional
+        The starting point in the complex plane.  The default is 1+0j.
+
+    Returns
+    -------
+    f : CZT
+        callable object ``f(x, axis=-1)`` for computing the chirp z-transform
+        on `x`
+
+    See Also
+    --------
+    ZoomFFT : for a friendly interface to partial FFT calculations
+
+    Notes
+    -----
+    If `w` does not lie on the unit circle, then the transform will be
+    around a spiral with exponentially-increasing radius.  Regardless,
+    angle will increase linearly.
+
+    For transforms that do lie on the unit circle, accuracy is better when
+    using `ZoomFFT`, since any numerical error in `w` is
+    accumulated for long data lengths, drifting away from the unit circle.
+
+    The chirp z-transform can be faster than an equivalent FFT with
+    zero padding.  Try it with your own array sizes to see.
+
+    However, the chirp z-transform is considerably less precise than the
+    equivalent zero-padded FFT, with differences on the order of 1e-7 from the
+    direct transform rather than 1e-15 as seen with zero-padding.
```

Comment:
There are some examples in https://github.com/scipy/scipy/issues/4288#issuecomment-75447314  comparing `czt(ones(N))` vs `fft(ones(N))`.  I made it better for the FFT case, but for the general case it's still limited by accumulation of error from the inaccurate step size.